ginisdk/src/androidTest/java/net/gini/android/authorization/EncryptedCredentialsStoreTest.java

```diff
@@ -0,0 +1,139 @@
+package net.gini.android.authorization;
+
+import static android.content.Context.MODE_PRIVATE;
+
+import android.content.SharedPreferences;
+import android.test.AndroidTestCase;
+
+/**
+ * Created by Alpar Szotyori on 08.10.2018.
+ *
+ * Copyright (c) 2018 Gini GmbH.
+ */
+public class EncryptedCredentialsStoreTest extends AndroidTestCase {
+
+    private SharedPreferences mSharedPreferences;
+    private EncryptedCredentialsStore mCredentialsStore;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mSharedPreferences = getContext().getSharedPreferences("GiniTests", MODE_PRIVATE);
+        mSharedPreferences.edit().clear().commit();
+
+        mCredentialsStore = new EncryptedCredentialsStore(mSharedPreferences, getContext());
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
+        mSharedPreferences.edit().clear().commit();
+    }
+
+    public void testEncryptsExistingPlaintextCredentials() {
+        // Given
+        final UserCredentials userCredentials = new UserCredentials("testuser@gini.net",
+                "12345678");
+        mCredentialsStore.mSharedPreferencesCredentialsStore.storeUserCredentials(userCredentials);
+        // When
+        mCredentialsStore.encryptExistingCredentials();
+        // Then
+        final UserCredentials encryptedUserCredentials =
+                mCredentialsStore.mSharedPreferencesCredentialsStore.getUserCredentials();
+        assertTrue(!userCredentials.getUsername().equals(encryptedUserCredentials.getUsername()));
+        assertTrue(!userCredentials.getPassword().equals(encryptedUserCredentials.getPassword()));
+    }
+
+    public void testDecryptsEncryptedExistingPlaintextCredentials() {
+        // Given
+        final UserCredentials userCredentials = new UserCredentials("testuser@gini.net",
+                "12345678");
+        mCredentialsStore.mSharedPreferencesCredentialsStore.storeUserCredentials(userCredentials);
+        // When
+        mCredentialsStore.encryptExistingCredentials();
+        // Then
+        final UserCredentials decryptedUserCredentials = mCredentialsStore.getUserCredentials();
+        assertEquals(userCredentials.getUsername(), decryptedUserCredentials.getUsername());
+        assertEquals(userCredentials.getPassword(), decryptedUserCredentials.getPassword());
+    }
+
+
+    public void testDoesNotEncryptAlreadyEncryptedCredentials() {
```

Comment:
By comparing the credentials I can see whether they have been encrypted twice because, if the second credentials are not the same as the first one it means, that the encrypted credentials have been encrypted again resulting in different encrypted values.

The `encryptExistingCredentials()` should encrypt only if there was no encryption (signaled by the lack of the encryption version in the shared preferences). If there was already an encryption (encryption version is present) the values are not encrypted again, the method should do nothing. That's why I wouldn't want to throw an exception.

Reply:
I should rename `encryptExistingCredentials()` to `encryptExistingPlaintextCredentials()`

Reply:
Yes, that's what I understood. Maybe just renaming the test case is enough, since they are encrypted but not stored.